llama_stack/core/distribution.py

```diff
@@ -264,9 +265,16 @@ def get_external_providers_from_module(
                         config_class="",
                     )
                 provider_type = provider.provider_type
-                # in the case we are building we CANNOT import this module of course because it has not been installed.
-                # return a partially filled out spec that the build script will populate.
-                registry[Api(provider_api)][provider_type] = spec
+                if isinstance(spec, list):
+                    # optionally allow people to pass inline and remote provider specs as a returned list.
+                    # with the old method, users could pass in directories of specs using overlapping code
+                    # we want to ensure we preserve that flexibility in this method.
+                    for adapter in spec:
```

Comment:
can you have three adapters? what if there is _no_ adapter? adapter implies remote right?

Reply:
but I guess since they both inherit from `ProviderSpec`, I can depend just on `provider_type` .